### PR TITLE
Replace lightgbm with PyTorch-based example to remove lightgbm dependency in visualization tutorial

### LINKED ISSUE #5255: New tutorial for visualization
### What is an issue?

In Optuna v3.5, [the visualization tutorial](https://optuna.readthedocs.io/en/v3.5.0/tutorial/10_key_features/005_visualization.html
) heavily depends on `LightGBMPruningCallback`.
After migrating the LightGBM integration (https://github.com/optuna/optuna/pull/5249), some contents of this tutorial, such as `plot_intermediate_values`, are no longer available.
Therefore, we want to prepare a new tutorial for visualization to demonstrate the functionalities of `optuna.visualization`.

The requirements for a new tutorial are as follows.
- **Do not depend on `optuna_integration`.**
- Demonstrates the following plot functions.
  - [plot_optimization_history()](https://optuna.readthedocs.io/en/v3.5.0/reference/visualization/generated/optuna.visualization.plot_optimization_history.html#optuna.visualization.plot_optimization_history)
  - [plot_intermediate_values()](https://optuna.readthedocs.io/en/v3.5.0/reference/visualization/generated/optuna.visualization.plot_intermediate_values.html#optuna.visualization.plot_intermediate_values)
  - [plot_parallel_coordinate()](https://optuna.readthedocs.io/en/v3.5.0/reference/visualization/generated/optuna.visualization.plot_parallel_coordinate.html#optuna.visualization.plot_parallel_coordinate)
  - [plot_contour()](https://optuna.readthedocs.io/en/v3.5.0/reference/visualization/generated/optuna.visualization.plot_contour.html#optuna.visualization.plot_contour)
  - [plot_slice()](https://optuna.readthedocs.io/en/v3.5.0/reference/visualization/generated/optuna.visualization.plot_slice.html#optuna.visualization.plot_slice)
  - [plot_param_importances()](https://optuna.readthedocs.io/en/v3.5.0/reference/visualization/generated/optuna.visualization.plot_param_importances.html#optuna.visualization.plot_param_importances)
  - [plot_edf()](https://optuna.readthedocs.io/en/v3.5.0/reference/visualization/generated/optuna.visualization.plot_edf.html#optuna.visualization.plot_edf)
  - [plot_rank()](https://optuna.readthedocs.io/en/v3.5.0/reference/visualization/generated/optuna.visualization.plot_rank.html#optuna.visualization.plot_rank)
  - [plot_timeline()](https://optuna.readthedocs.io/en/v3.5.0/reference/visualization/generated/optuna.visualization.plot_timeline.html#optuna.visualization.plot_timeline)
  - How to customize generated figures like the figure below. image

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

Resolve #5255

## Description of the changes

1. Replace the existing example using lightgbm with PyTorch since Optuna is a part of PyTorch ecosystem. 
The newly added PyTorch example comes from the multi-objective tutorial page: [Multi-objective Optimization with Optuna](https://optuna.readthedocs.io/en/stable/tutorial/20_recipes/002_multi_objective.html). I've removed a few hyperparameters so as not to make the figure less clear due to overlapping text. 
2. Revert the removal of two sections by #5249

### CODE REVIEW

**Comment by nabenabe0928** [comment]
@y0z Could you review this PR?

**Review by y0z — APPROVED** [review]
Thank you for your contribution.
LGTM!

NIT: Although coincidentally, in the example, the first trial achieves the best accuracy. To better demonstrate the optimization history plot, another random seed involving best-value updates might be preferable.

image

**Comment by nzw0301** [comment]
@y0z Thank you for your review and suggestion. How about this plot that I used a different random seed?
 
newplot

**Comment by y0z** [comment]
@nzw0301 

Thank you for your update! LGTM 👏

**Review by nabenabe0928 — APPROVED** [review]
Thank you for the PR, LGTM!
